service/history/history_engine2_test.go

```diff
@@ -1361,6 +1362,28 @@ func (s *engine2Suite) TestStartWorkflowExecution_Dedup_Running_SameRequestID()
 	s.Equal(s.tv.RunID(), resp.GetRunId())
 }
 
+func (s *engine2Suite) TestStartWorkflowExecution_Dedup_Running_SameAttachedRequestID() {
+	// no error when request ID is the same
+	s.setupStartWorkflowExecutionForRunning()
+	startRequest := makeMockStartRequest(
+		s.tv,
+		enumspb.WORKFLOW_ID_REUSE_POLICY_UNSPECIFIED,
+		enumspb.WORKFLOW_ID_CONFLICT_POLICY_TERMINATE_EXISTING,
+	)
+
+	startRequest.StartRequest.RequestId = s.tv.String("AttachedRequestID1")
+	resp, err := s.historyEngine.StartWorkflowExecution(metrics.AddMetricsContext(context.Background()), startRequest)
+	s.NoError(err)
+	s.True(resp.Started)
+	s.Equal(s.tv.RunID(), resp.GetRunId())
+
+	startRequest.StartRequest.RequestId = s.tv.String("AttachedRequestID2")
+	resp, err = s.historyEngine.StartWorkflowExecution(metrics.AddMetricsContext(context.Background()), startRequest)
+	s.NoError(err)
+	s.True(resp.Started)
```

Comment:
Maybe I'm not following this but shouldn't `Started` be false here since we're attaching to an existing workflow?

Reply:
Oh, yes! Nice catch!